### PR TITLE
Server rates

### PR DESCRIPTION
**Pull request by danyaljj**
- Auto-format the pipeline. 
 - Add the server rate.

### CODE REVIEW

File: pipeline/src/main/java/edu/illinois/cs/cogcomp/pipeline/common/PipelineConfigurator.java [hunk 1]
```diff
@@ -51,35 +51,37 @@
             SrlConfigurator.INSTANTIATE_PREPROCESSOR.key, FALSE);
 
     /**
-     * if 'true', the PipelineFactory will return a sentence-level pipeline that will use all viable annotators in
```
Comment (danyaljj) [inline pipeline/src/main/java/edu/illinois/cs/cogcomp/pipeline/common/PipelineConfigurator.java]: the only changes to this file is auto-formatting.

File: pipeline/src/main/java/edu/illinois/cs/cogcomp/pipeline/main/PipelineFactory.java [hunk 2]
```diff
@@ -48,8 +48,8 @@
     private static Logger logger = LoggerFactory.getLogger(PipelineFactory.class);
 
     /**
-     * create an AnnotatorService with the given view names in the argument. The names are supposed be strings,
```
Comment (danyaljj) [inline pipeline/src/main/java/edu/illinois/cs/cogcomp/pipeline/main/PipelineFactory.java]: the only changes to this file is auto-formatting.

File: pipeline/src/main/java/edu/illinois/cs/cogcomp/pipeline/main/SentencePipeline.java [hunk 3]
```diff
@@ -20,13 +20,14 @@
 import java.util.Set;
 
 /**
- * Where possible, run {@link Annotator} members over one sentence at a time and splice their outputs together.
```
Comment (danyaljj) [inline pipeline/src/main/java/edu/illinois/cs/cogcomp/pipeline/main/SentencePipeline.java]: the only changes to this file is auto-formatting.

File: pipeline/src/main/java/edu/illinois/cs/cogcomp/pipeline/server/MainServer.java [hunk 4]
```diff
@@ -25,6 +25,7 @@
 import java.io.IOException;
 import java.io.UnsupportedEncodingException;
```
Comment (danyaljj) [inline pipeline/src/main/java/edu/illinois/cs/cogcomp/pipeline/server/MainServer.java]: changes to rates are here.

File: pipeline/src/main/java/edu/illinois/cs/cogcomp/pipeline/server/MainServer.java [hunk 5]
```diff
@@ -77,23 +96,58 @@ public static void startServer(String[] args, Logger logger) {
 
         port(parseResults.getInt("port"));
 
+        // create a hashmap to keep track of client ip addresses and their
+        int rate = parseResults.getInt("rate");
+        if( rate > 0) {
+            clients = new HashMap<String, Integer>();
```
Comment (bhargav) [inline pipeline/src/main/java/edu/illinois/cs/cogcomp/pipeline/server/MainServer.java]: If SparkJava handles requests concurrently, you should use a ConcurrentHashMap here.

File: pipeline/README.md [hunk 6]
```diff
@@ -304,11 +304,12 @@ pipeline/scripts/runWebserver.sh
 
 The following arguments are supported:
 ```shell
-usage: pipeline/scripts/runWebserver.sh [-h] [--port PORT]
+usage: pipeline/scripts/runWebserver.sh [-h] [--port PORT] [--rate HOURS]
```
Comment (bhargav) [inline pipeline/README.md]: HOURS? This parameter is the number of queries per day right?

Comment (danyaljj) [inline pipeline/README.md]: oops.

File: pipeline/src/main/java/edu/illinois/cs/cogcomp/pipeline/server/MainServer.java (hunk 5 above)
Comment (danyaljj) [inline pipeline/src/main/java/edu/illinois/cs/cogcomp/pipeline/server/MainServer.java]: Good point.

**Comment by danyaljj** [comment]
Addressed the comments.

**Pull request merged**